contracts/InterchainTokenFactory.sol

```diff
@@ -153,8 +153,7 @@ contract InterchainTokenFactory is IInterchainTokenFactory, ITokenManagerType, M
             token.transferDistributorship(distributor);
 
             tokenManager.removeFlowLimiter(address(this));
-
-            if (distributor != address(0)) tokenManager.addFlowLimiter(distributor);
+            tokenManager.addFlowLimiter(distributor);
```

Comment:
since there can be any number of flow limiters this does not really matter i don't think.